Support JDBC validate connection (#24127)
trinodb/trino: docs/src/main/sphinx/client/jdbc.md

    `PREPARE <statement>` followed by `EXECUTE <statement>`. This reduces
    network overhead and uses smaller HTTP headers and requires Trino 431 or
    greater.
* - `validateConnection`

>   - Defaults to `false`. If set to `true`, connectivity and credential will be 

Author: Fixed.

    Valid values are JSON with Zstandard compression, `json+zstd` (recommended),
    JSON with LZ4 compression `json+lz4`, and uncompressed JSON `json`. By
    default, the default encoding configured on the cluster is used.
* - `validateConnection`

>   - Defaults to `false`. If set to `true`, connectivity and credentials will be 

Reviewer: ```suggestion
  - Defaults to `false`. If set to `true`, connectivity and credentials are
```

>     validated when create a connection, and when `java.sql.Connection.isValid(int)` 

Reviewer: ```suggestion
    validated when the connection is created, and when `java.sql.Connection.isValid(int)` 
```
